### DISCUSSION TITLE
1.0 Plan

### DISCUSSION THREAD

**Discussion by johnsoncodehk** [vuejs/language-tools#168] [General]
Hi user, hope you are satisfied with Volar. :)

I just released [vscode-vue-languageservice](https://www.npmjs.com/package/vscode-vue-languageservice), and the current phased work has been completed. Let me briefly introduce Volar here, and then talk about future plans.

> What is Volar?

Volar is [a great song](https://youtu.be/67a_sfCeZW8) by my favorite singer!

> What is Volar's goal?

1. Realize the `*.ts` same level of coding experience in `<script>`.
2. Maximize the use of SFC design to provide language features. (Not only combine HTML, Css, Js and provide language features separately)

> What are the features of Volar?

Volar provides Vue 3 language support, as well as some features that I think are helpful. E.g:

- Template type-checking
- Pug <-> HTML conversion
- Split `<template>`, `<script>`, `<style>` into independent editors
- TS Plugin
- Css class name renaming, find references, goto definition
- ...

## IDE support other than VSCode

I do not plan to maintain IDEs other than VSCode, which is too much work for me. Secondly, if I maintain an IDE that I don't use, it will be difficult for me to find the problem and fix it early.

If you are an extension developer of other IDEs, you can use vscode-vue-languageservice for your IDE to create extensions with Volar language features. vscode-vue-languageservice is a general-purpose module for Language Server. It has the same API as vscode-css-languageservice and vscode-html-languageservice. It can be used in any IDE in theory.

## Future plan

The biggest challenge for Volar in the future is to implement the Plugin API, which can change the type checking behavior of `vue-tsc` and Volar, and provide additional language support to support third-party plug-ins such as i18n.

The current architecture of vscode-vue-languageservice does not support the implementation of Plugin API, so I plan to completely rewrite vscode-vue-languageservice to support Plugin API in the future. This is a very large workload. I can only develop Volar in my spare time, so This work cannot be carried out for the time being.

I do hope to develop Volar full-time to implement the Plugin API. If you want to see this happen, please consider [becoming a sponsor](https://github.com/sponsors/johnsoncodehk). Thank you!

---

Thanks a lot to early users who reported a lot of bugs!

@pgy108103, @cereschen, @kadiryazici

**Comment by cereschen** [comment]
We are witnessing volar getting better and better :tada:
